Trimble.Modus.Components/Controls/TMDataGrid/DataGrid.xaml.cs

```diff
@@ -0,0 +1,727 @@
+            {
+                SelectedItem = null;
+                SelectedItems.Clear();
+            }
+        }
+    }
+
+    #endregion
+
+    #region Properties
+
+    public List<int> ReadOnlyList
+    {
+        get => (List<int>)GetValue(ReadOnlyListProperty);
+        set => SetValue(ReadOnlyListProperty, value);
+    }
+
+    /// <summary>
+    /// Set if divider should be shown
+    /// </summary>
+    public bool ShowDivider
+    {
+        get => (bool)GetValue(ShowDividerProperty);
+        set => SetValue(ShowDividerProperty, value);
+    }
+
+    /// <summary>
+    /// ItemSizingStrategy
+    /// Default Value is MeasureFirstItem, except on Android
+    /// </summary>
+    public ItemSizingStrategy ItemSizingStrategy
+    {
+        get => (ItemSizingStrategy)GetValue(ItemSizingStrategyProperty);
+        set => SetValue(ItemSizingStrategyProperty, value);
+    }
+
+    /// <summary>
+    /// ItemsSource of the DataGrid
+    /// </summary>
+    public IEnumerable ItemsSource
+    {
+        get => (IEnumerable)GetValue(ItemsSourceProperty);
+        set => SetValue(ItemsSourceProperty, value);
+    }
+
+    private IList<object>? _internalItems;
+
+    internal IList<object>? InternalItems
+    {
+        get => _internalItems;
+        set
+        {
+            _internalItems = value;
+
+            _collectionView.ItemsSource = _internalItems;
+        }
+    }
+
+    /// <summary>
+    /// Columns
+    /// </summary>
+    public ObservableCollection<DataGridColumn> Columns
+    {
+        get => (ObservableCollection<DataGridColumn>)GetValue(ColumnsProperty);
+        set => SetValue(ColumnsProperty, value);
+    }
+
+    /// <summary>
+    /// Gets or sets if the grid is sortable. Default value is true.
+    /// Sortable columns must implement <see cref="IComparable"/>
+    /// If you want to enable or disable sorting for specific column please use <c>SortingEnabled</c> property
+    /// </summary>
+    public bool IsSortable
+    {
+        get => (bool)GetValue(IsSortableProperty);
+        set => SetValue(IsSortableProperty, value);
+    }
+
+    /// <summary>
+    /// Enables selection in dataGrid. Default value is True
+    /// </summary>
+    public SelectionMode SelectionMode
+    {
+        get => (Microsoft.Maui.Controls.SelectionMode)GetValue(SelectionModeProperty);
+        set => SetValue(SelectionModeProperty, value);
+    }
+
+    /// <summary>
+    /// Selected item
+    /// </summary>
+    public object? SelectedItem
+    {
+        get => GetValue(SelectedItemProperty);
+        set => SetValue(SelectedItemProperty, value);
+    }
+    /// <summary>
+    /// Selected item
+    /// </summary>
+    public IList<object> SelectedItems
+    {
+        get => (IList<object>)GetValue(SelectedItemsProperty);
+        set => SetValue(SelectedItemsProperty, value);
+    }
+    /// <summary>
+    /// Column index and sorting order for the DataGrid
+    /// </summary>
+    public SortData SortedColumnIndex
```

Comment:
The name and the type doesn't match. SortedColumnIndex feels like it is an int but it is not.

Reply:
Renamed